src/Symfony/Bundle/FrameworkBundle/Resources/config/annotations.xml

```diff
@@ -7,20 +7,6 @@
     <services>
         <defaults public="false" />
 
-        <service id="annotations.reader" class="Doctrine\Common\Annotations\AnnotationReader">
-            <call method="addGlobalIgnoredName">
-                <argument>required</argument>
-                <argument type="service">
-                    <!-- dummy arg to register class_exists as annotation loader only when required -->
-                    <service class="Doctrine\Common\Annotations\AnnotationRegistry">
-                        <call method="registerLoader">
```

Comment:
you should keep this definition, and just replace registerLoader by registerUniqueLoader here

THEN
in FrameworkExtension.php you need to fetch the "annotations.reader" definition using `$container->getDefinition('annotations.reader')`, and tweak it as needed to replace the call by "registerLoader" when the previous method doesn't exist.

Reply:
Something like:
`$container->getDefinition('annotations.reader')->getArgument(1)->setMethodCalls(array('registerLoader', array('class_exists')));`

Reply:
Ah ok, thanks. I would never have thought of that. I will make that change now.

Reply:
Sorry @nicolas-grekas this is beyond my skill and understanding. `$container->getDefinition('annotations.reader')
    ->getMethodCalls()[0][1][1]` returns a `Symfony\Component\DependencyInjection\Reference` which doesn't have a `setMethodCalls` method. That method does exist on the `Definition` but I'm not sure how to duplicate the tiered structure in the XML or the call to a service.

Reply:
Instead of an inlined service, you may move the service on L15 as a named one. That may help you, and future readers.

Reply:
Looked back at the original code and doing this as a configurator was the suggested approach.  Since I couldn't figure out how to do this another way I did it with a configurator. This has the added benefit of calling at static methods statically and not relying oh PHPs poor error control around calling them as instance methods.
 I wasn't sure where to put the class though. I didn't want to create a special namespace just for it, so I shoehorned it into CacheWarmer which is probably doesn't belong.

Reply:
That's reasonable, but actually we made it this way on purpose: to not add yet another class just for this.
I just pushed the change that makes this works as suggested :)